### ISSUE TITLE
Require Authorize on Hubs as a Global Option

### ISSUE THREAD

**Issue by mikaelm12** [aspnet/SignalR#2480]
I was thinking we could introduce a global hub option for requiring auth on all hubs. There was a way to do this in classic SignalR but for ASP.NET Core SignalR you'd have to add the `Authorize` tag to each hub.

It could look something like 
```csharp
services.AddSignalR(options => 
{
    options.RequireAuthorization = true
})
```

**Comment by davidfowl** [comment]
Yea that API might be a bit too simplistic. We may want to look at the hub pipeline modules again.

**Comment by mikaelm12** [comment]
Yeah, that's fine. Was throwing something out there to start the conversation. Just wanted us to take a look at the idea.
